programs/sbf/tests/programs.rs

```diff
@@ -518,6 +552,138 @@ fn test_sol_alloc_free_no_longer_deployable() {
     assert!(bank.process_transaction(&invoke_tx).is_ok());
 }
 
+#[test]
+#[cfg(feature = "sbf_rust")]
+#[should_panic(
+    expected = "called `Result::unwrap()` on an `Err` value: TransactionError(InstructionError(1, InvalidAccountData))"
+)]
+fn test_sol_alloc_free_no_longer_deployable_with_upgradeable_loader() {
+    solana_logger::setup();
+
+    let GenesisConfigInfo {
+        genesis_config,
+        mint_keypair,
+        ..
+    } = create_genesis_config(50);
+
+    let (bank, bank_forks) = Bank::new_with_bank_forks_for_tests(&genesis_config);
+    let mut bank_client = BankClient::new_shared(bank.clone());
+
+    // Populate loader account with `solana_sbf_rust_deprecated_loader` elf, which
+    // depends on `sol_alloc_free_` syscall. This can be verified with
+    // $ elfdump solana_sbf_rust_deprecated_loader.so
+    // : 0000000000001ab8  000000070000000a R_BPF_64_32            0000000000000000 sol_alloc_free_
+    // In the symbol table, there is `sol_alloc_free_`.
+    // In fact, `sol_alloc_free_` is called from sbf allocator, which is originated from
+    // AccountInfo::realloc() in the program code.
+    let program_buffer_keypair = Keypair::new();
+    let program_keypair = Keypair::new();
+    let authority_keypair = Keypair::new();
+
+    // Expect that deployment to fail. B/C during deployment, there is an elf
+    // verification step, which uses the runtime to look up relocatable symbols
+    // in elf inside syscall table. In this case, `sol_alloc_free_` can't be
+    // found in syscall table. Hence, the verification fails and the deployment
+    // fails.
+    let (_bank, _program_id) = load_upgradeable_program_and_advance_slot(
+        &mut bank_client,
+        bank_forks.as_ref(),
+        &mint_keypair,
+        &program_buffer_keypair,
+        &program_keypair,
+        &authority_keypair,
+        "solana_sbf_rust_deprecated_loader",
+    );
+}
+
+#[test]
+#[cfg(feature = "sbf_rust")]
+fn test_sol_alloc_free_deployable_with_upgradeable_loader() {
+    solana_logger::setup();
+
+    let GenesisConfigInfo {
+        mut genesis_config,
+        mint_keypair,
+        ..
+    } = create_genesis_config(50);
+
+    // deactivate `disable_deploy_of_alloc_free_syscall` feature so that the program
+    // that depends on sol_alloc_free_ can be deployed and executed.
+    genesis_config
+        .accounts
+        .remove(&solana_sdk::feature_set::disable_deploy_of_alloc_free_syscall::id());
```

Comment:
It seems this feature is already activated on mainnet-beta. Do we want to keep the test where it's disabled?

Reply:
I vote for no. Just keep `test_sol_alloc_free_no_longer_deployable_with_upgradeable_loader` and remove `test_sol_alloc_free_deployable_with_upgradeable_loader`.

Reply:
@Lichtso WDYT?

Reply:
removed.